view_test.go

```diff
@@ -214,6 +252,43 @@ func TestStop_stopsPolling(t *testing.T) {
 	}
 }
 
+func TestStop_stopsFetchWithCancel(t *testing.T) {
+	ctx, ctxCancel := context.WithCancel(context.Background())
+	view := newView(&newViewInput{
+		Dependency: &dep.FakeDepBlockingQuery{
+			Name:          "ctxCancel",
+			BlockDuration: time.Duration(5 * time.Minute),
+			Ctx:           ctx,
+		},
+	})
+	view.ctxCancel = ctxCancel
+
+	doneCh := make(chan struct{})
+	successCh := make(chan struct{})
+	errCh := make(chan error)
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		view.fetch(doneCh, successCh, errCh)
+		wg.Done()
+	}()
+
+	view.stop()
+
+	select {
+	case <-doneCh:
+		t.Errorf("unexpected doneCh")
+	case <-successCh:
+		t.Errorf("unexpected successCh")
+	case <-errCh:
+		t.Errorf("unexpected errCh")
+	case <-time.After(5 * time.Millisecond):
+		// Successfully stopped by context
+		wg.Wait()
```

Comment:
looking at these two tests, the wg.Wait() is in different places for the tests. just want to double-check if there is something deliberate in the placement of wg.Wait() that's gone over my head.

Reply:
Hm you're right they're not consistent. Thanks Lorna! I can update this to move `wg.Wait()` outside like the other test. The `<-time.After` is mainly to prove that nothing are in the other channels since `select` is random and not serial for the cases.

Reply:
👍 not urgent. thanks for checking me on that!